src/all/namicomi/src/eu/kanade/tachiyomi/extension/all/namicomi/NamicomiFilters.kt

```diff
@@ -0,0 +1,289 @@
+            Tag("reverse-harem", intl["theme_reverse_harem"]),
+            Tag("samurai", intl["theme_samurai"]),
+            Tag("school-life", intl["theme_school_life"]),
+            Tag("supernatural", intl["theme_supernatural"]),
+            Tag("survival", intl["theme_survival"]),
+            Tag("time-travel", intl["theme_time_travel"]),
+            Tag("traditional-games", intl["theme_traditional_games"]),
+            Tag("vampires", intl["theme_vampires"]),
+            Tag("video-games", intl["theme_video_games"]),
+            Tag("villainess", intl["theme_villainess"]),
+            Tag("virtual-reality", intl["theme_virtual_reality"]),
+            Tag("zombies", intl["theme_zombies"]),
+        )
+
+        return tags.sortIfTranslated(intl)
+    }
+
+    // to get all tags from dex https://api.mangadex.org/title/tag
+    internal fun getTags(intl: Intl): List<Tag> {
+        return getContents(intl) + getFormats(intl) + getGenres(intl) + getThemes(intl)
+    }
+
+    private data class TagMode(val title: String, val value: String) {
+        override fun toString(): String = title
+    }
+
+    private fun getTagModes(intl: Intl) = arrayOf(
+        TagMode(intl["mode_and"], "and"),
+        TagMode(intl["mode_or"], "or"),
+    )
+
+    private class TagInclusionMode(intl: Intl, modes: Array<TagMode>) :
+        Filter.Select<TagMode>(intl["included_tags_mode"], modes, 0),
+        UrlQueryFilter {
+
+        override fun addQueryParameter(url: HttpUrl.Builder, extLang: String) {
+            url.addQueryParameter("includedTagsMode", values[state].value)
+        }
+    }
+
+    private class TagExclusionMode(intl: Intl, modes: Array<TagMode>) :
+        Filter.Select<TagMode>(intl["excluded_tags_mode"], modes, 1),
+        UrlQueryFilter {
+
+        override fun addQueryParameter(url: HttpUrl.Builder, extLang: String) {
+            url.addQueryParameter("excludedTagsMode", values[state].value)
+        }
+    }
+
+    private class TagsFilter(intl: Intl, innerFilters: FilterList) :
+        Filter.Group<Filter<*>>(intl["tags_mode"], innerFilters),
+        UrlQueryFilter {
+
+        override fun addQueryParameter(url: HttpUrl.Builder, extLang: String) {
+            state.filterIsInstance<UrlQueryFilter>()
+                .forEach { filter -> filter.addQueryParameter(url, extLang) }
+        }
+    }
+
+    private fun getTagFilters(intl: Intl): FilterList = FilterList(
+        TagInclusionMode(intl, getTagModes(intl)),
+        TagExclusionMode(intl, getTagModes(intl)),
+    )
+
+    internal fun addFiltersToUrl(url: HttpUrl.Builder, filters: FilterList, extLang: String): HttpUrl {
+        filters.filterIsInstance<UrlQueryFilter>()
+            .forEach { filter -> filter.addQueryParameter(url, extLang) }
+
+        return url.build()
+    }
+
+    private fun List<Tag>.sortIfTranslated(intl: Intl): List<Tag> = apply {
+        if (intl.chosenLanguage == NamicomiConstants.english) {
```

Comment:
english is used to indicate any translation?

Reply:
The tag lists are already sorted by the English names in the code. So, if the English extension source is loaded we don't need to bother sorting it again. The logic was copied from the MangaDex ext.
For other translations, we need to sort the localized strings.